content/200-concepts/100-components/02-prisma-client/260-client-extensions/120-query.mdx

```diff
@@ -1,16 +1,168 @@
 ---
 title: 'Create custom Prisma Client queries'
-metaTitle: 'Prisma Client extensions (Preview)'
+metaTitle: 'Prisma Client extensions: query component (Preview)'
 metaDescription: 'Extend the functionality of Prisma Client'
 tocDepth: 3
 ---
 
 <TopBlock>
 
-<Admonition type="warning">
+From version 4.7.0, you can use the `query` [Prisma Client extensions](/concepts/components/prisma-client/client-extensions) component type to hook into the query life-cycle and modify an incoming query or its result.
 
-We plan to add `query` extensions to a future version of Prisma. This page is a placeholder.
-
-</Admonition>
+You can use the `query` extension component as an alternative to [middlewares](/concepts/components/prisma-client/middleware#performance-and-appropriate-use-cases). Unlike middlewares, the `query` extension component gives you [user isolation](/concepts/components/prisma-client/client-extensions#extended-clients), such as in a row-level security (RLS) scenario, and end-to-end type safety. [Learn more](#query-extensions-versus-middlewares).
 
 </TopBlock>
+
+## Enable the preview feature
+
+Before you create Prisma Client extensions, you must enable the `clientExtensions` feature flag in the `generator` block of your `schema.prisma` file, as follows:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["clientExtensions"]
+}
+```
+
+## Create custom queries
```

Comment:
Calling these only "custom queries" might not be broad enough? Yes, you can _add_ queries, but you can also modify all existing queries, right? Should we somehow mention and highlight that as well?

Reply:
I'll change the wording in the doc so that it talks about adding or modifying queries instead of just adding